Add register|unregister hook console command

PrestaEdit opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add the `registerHook` and `unregisterHook` action on `prestashop:module` command.
| Type?             | new feature
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27528.
| How to test?      | Use a console command indicate further in this issue.
| Possible impacts? | n/d

Available command to test:

- `./bin/console prestashop:module registerHook ps_bestsellers --hook=adminStatsModules --hook=displayAdminCustomers`
- `./bin/console prestashop:module registerHook ps_bestsellers --hook=adminStatsModules`
- `./bin/console prestashop:module unregisterHook ps_bestsellers --hook=adminStatsModules --hook=displayAdminCustomers`
- `./bin/console prestashop:module unregisterHook ps_bestsellers --hook=adminStatsModules`


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/27531)
<!-- Reviewable:end -->

Commit:
Remove "select" case and fix "editable" one in HelperList

PrestaEdit committed:
Add register|unregister hook console command

mickaelandrieu commented on src/Adapter/Module/Module.php:
```diff
@@ -366,6 +368,34 @@ public function onReset(): bool
+            return false;
+        }
+
+        $result = $this->instance->unregisterHook($hookName);
```
```diff
+ return $this->instance->unregisterHook($hookName);
- $result = $this->instance->unregisterHook($hookName);
- return $result;
```

mickaelandrieu replied:
same for above and (maybe) cast to bool if required ? Are we sure that `unregisterHook()` function returns a boolean ?

mickaelandrieu commented on src/PrestaShopBundle/Command/ModuleCommand.php:
```diff
@@ -106,7 +108,8 @@ protected function configure()
             ->addArgument('module name', InputArgument::REQUIRED, 'Module on which the action will be executed')
-            ->addArgument('file path', InputArgument::OPTIONAL, 'YML file path for configuration');
+            ->addArgument('file path', InputArgument::OPTIONAL, 'YML file path for configuration')
+            ->addOption('hook', null, InputOption::VALUE_IS_ARRAY|InputOption::VALUE_OPTIONAL, 'Hook name', []);
```
This breaks because you forgot to import InputOption from the Symfony Component

PrestaEdit replied:
Nice catch ; have tested this on an another server than my local commits. Not a good idea, :)

PrestaEdit committed:
Add missing use (InputOption)

PrestaEdit committed:
Improve the method return

PrestaEdit committed:
Add missing third params ($options) on method

PrestaEdit committed:
PHP CS fix

mickaelandrieu commented on src/PrestaShopBundle/Command/ModuleCommand.php:
```diff
@@ -183,9 +195,15 @@ protected function executeConfigureModuleAction($moduleName, $file = null)
     }
 
-    protected function executeGenericModuleAction($action, $moduleName)
+    protected function executeGenericModuleAction($action, $moduleName, $options = null)
```
I'm wondering ... what if the default value of $options is an empty array ? This make the function easier as we can remove the if/else statment : WDYT ?

PrestaEdit replied:
As the call is not he same (or we need to edit all actions to have the $options params for nothing), you still need to make an if statement or you will have a second parameters send to action that expect only one.

PrestaEdit committed:
PHP Cs fix

matks commented on src/Core/Addon/Module/ModuleManager.php:
```diff
@@ -580,6 +580,70 @@ public function reset($name, $keep_data = false)
+     * Register a module on given hook.
+     *
+     * @param string $name The module name
+     * @param array $hooksName The hooks name
```
if there is many should'nt be it `$hookNames` or `$hooksNames` ?

matks commented on src/Core/Addon/Module/ModuleManager.php:
```diff
@@ -580,6 +580,70 @@ public function reset($name, $keep_data = false)
+     *
+     * @return bool True for success
+     */
+    public function registerHook($name, $hooksName)
```
You can typehint

matks commented on src/Core/Addon/Module/ModuleManager.php:
```diff
@@ -580,6 +580,70 @@ public function reset($name, $keep_data = false)
+                $result &= $module->onRegisterHook($hookName);
+            }
+        } catch (Exception $e) {
+            throw new Exception($this->translator->trans('Error when register hook on module %module%. %error_details%', ['%module%' => $name, '%error_details%' => $e->getMessage()], 'Admin.Modules.Notification'), 0, $e);
```
```suggestion
            throw new Exception($this->translator->trans('Error when registering hook on module %module%. %error_details%', ['%module%' => $name, '%error_details%' => $e->getMessage()], 'Admin.Modules.Notification'), 0, $e);
```

matks commented on src/Core/Addon/Module/ModuleManager.php:
```diff
@@ -580,6 +580,70 @@ public function reset($name, $keep_data = false)
+                $result &= $module->onUnregisterHook($hookName);
+            }
+        } catch (Exception $e) {
+            throw new Exception($this->translator->trans('Error when unregister hook on module %module% . %error_details%', ['%module%' => $name, '%error_details%' => $e->getMessage()], 'Admin.Modules.Notification'), 0, $e);
```
```suggestion
            throw new Exception($this->translator->trans('Error when unregistering hook on module %module% . %error_details%', ['%module%' => $name, '%error_details%' => $e->getMessage()], 'Admin.Modules.Notification'), 0, $e);
```

matks reviewed (commented):
Idea looks OK and code is overall OK. I hope we can test this but it seems hard to setup.

PierreRambaud commented on src/Core/Addon/Module/ModuleManager.php:
```diff
@@ -580,6 +580,70 @@ public function reset($name, $keep_data = false)
+            throw new Exception($this->translator->trans('Error when unregister hook on module %module% . %error_details%', ['%module%' => $name, '%error_details%' => $e->getMessage()], 'Admin.Modules.Notification'), 0, $e);
+        }
+
+        $this->checkAndClearCache($result);
```
```suggestion
        $this->checkAndClearCache((bool) $result);
```

Otherwise, it's an int

matks reviewed (changes requested):
Still 2 feedbacks not adressed ;)

PierreRambaud commented on src/Adapter/Module/Module.php:
```diff
@@ -47,6 +47,8 @@ class Module implements ModuleInterface
     public const ACTION_DISABLE_MOBILE = 'disable_mobile';
     public const ACTION_RESET = 'reset';
     public const ACTION_UPGRADE = 'upgrade';
+    public const ACTION_REGISTER_HOOK = 'registerHook';
```
Keep the snake case to be consistent with what you have above? :thinking:

PierreRambaud reviewed (commented):
One comment about the constant value

Progi1984 commented on src/Adapter/Module/Module.php:
```diff
@@ -366,6 +368,30 @@ public function onReset(): bool
+    /**
+     * {@inheritdoc}
+     */
+    public function onRegisterHook($hookName): bool
```
```suggestion
    public function onRegisterHook(string $hookName): bool
```

Progi1984 commented on src/Adapter/Module/Module.php:
```diff
@@ -366,6 +368,30 @@ public function onReset(): bool
+    /**
+     * {@inheritdoc}
+     */
+    public function onUnregisterHook($hookName): bool
```
```suggestion
    public function onUnregisterHook(string $hookName): bool
```

PierreRambaud replied:
These methods must be added to the interface!

Progi1984 commented on src/Core/Addon/Module/ModuleManager.php:
```diff
@@ -580,6 +580,70 @@ public function reset($name, $keep_data = false)
+     *
+     * @return bool True for success
+     */
+    public function registerHook(string $name, array $hooksNames)
```
```suggestion
    public function registerHook(string $name, array $hooksNames): bool
```

Progi1984 commented on src/Core/Addon/Module/ModuleManager.php:
```diff
@@ -580,6 +580,70 @@ public function reset($name, $keep_data = false)
+     *
+     * @return bool True for success
+     */
+    public function unregisterHook(string $name, array $hooksNames)
```
```suggestion
    public function unregisterHook(string $name, array $hooksNames): bool
```

Progi1984 commented on src/PrestaShopBundle/Command/ModuleCommand.php:
```diff
@@ -183,9 +195,15 @@ protected function executeConfigureModuleAction($moduleName, $file = null)
     }
 
-    protected function executeGenericModuleAction($action, $moduleName)
+    protected function executeGenericModuleAction($action, $moduleName, $options = null)
```
```suggestion
    protected function executeGenericModuleAction($action, $moduleName, array $options = null)
```

PrestaEdit replied:
We can't do that, as the check is make to see if we have an array to pass it to the action.

Progi1984 reviewed (commented):
Add some typehint

PrestaEdit committed:
Update src/Adapter/Module/Module.php

Co-authored-by: Progi1984 <progi1984@gmail.com>

PrestaEdit committed:
Add methods to interface

PrestaEdit committed:
Update src/Adapter/Module/Module.php

Co-authored-by: Progi1984 <progi1984@gmail.com>

PrestaEdit commented:
Hi everyone,

I made somes changes. I'd like to make a squash commits as 10+ commits for this is just not possible.
It's seems not possible due to a merge commit inside them.

I let you address the last request changes and once this is ready to be merge, a newer Pull Request, cleaner, will be done.

Progi1984 reviewed (changes requested):
@PrestaEdit Could you rebase your PR and remove your merge commits ?

PrestaEdit commented:
Hi everyone,

I try to make this today.

kpodemski commented:
Hi @PrestaEdit 

Any chance to finish this PR? It could be great to have it in the Core

PrestaEdit commented:
Rebase done.

kpodemski reviewed (changes requested):
Hello @PrestaEdit 

Some checks in CI are failing. Could you please check them?

RomPiot committed:
fix: set the product redirection to another product (301 & 302)

ludoviccardinale committed:
Check overrides

ShaiMagal committed:
Update Customer.php

matks committed:
Move the array_unshift in Product::setWsPositionInCategory() to avoid Trying to access array offset on value of type null

matthieu-rolland committed:
refacto conflict verification + integration tests

matthieu-rolland committed:
exclude module tests file in php-cs-fixer config

matthieu-rolland committed:
test

matthieu-rolland committed:
add composer commands for php-cs-fixer and phpstan

matthieu-rolland committed:
rename ModuleOverrideVerifier into ModuleOverrideChecker

matthieu-rolland committed:
pass ps override dir as parameter to overrideChecker service

Codencode committed:
Update AdminCartsController.php

matthieu-rolland committed:
fix wrong condition on override check at module enable

matthieu-rolland committed:
re-order composer commands correctly

matthieu-rolland committed:
add unit tests

matthieu-rolland committed:
add check on constants and properties + tests

matthieu-rolland committed:
refactor method organization

matthieu-rolland committed:
replace token detection with regex

Codencode committed:
Added management of the 'addedByModule' field in the 'CustomizationField' object

In the ProductFormDataProvider::extractCustomizationsData() method, I added the 'addedByModule' key to the $fields array.
In the CustomizationFieldsCommandsBuilder::buildCustomizationFields() method, I added the 'added_by_module' key to the $customizationFields array.

matthieu-rolland committed:
Update default catalog 8.2.x

matthieu-rolland committed:
changelog 8.2

matthieu-rolland committed:
add new module hook filter service

matthieu-rolland committed:
allow fetching front container from ContainerFinder

Hlavtox committed:
Initial commit

matthieu-rolland committed:
fix service availability issue

jolelievre committed:
Revert unnecessary changes from merge action

Hlavtox committed:
Add more options to pagination

jolelievre committed:
Test fast login in UI tests

Hlavtox committed:
Remove useless pack quantity

Hlavtox committed:
Remove unnecessary cart present

Hlavtox committed:
Unify error handling

Progi1984 committed:
Fixed CI

dependabot[bot] committed:
build(deps-dev): bump @types/mocha from 10.0.9 to 10.0.10 in /tests/UI

Bumps [@types/mocha](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/mocha) from 10.0.9 to 10.0.10.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/mocha)

---
updated-dependencies:
- dependency-name: "@types/mocha"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

kpodemski committed:
Fix obvious notice in Warehouse class

fix getting shop group id

fix

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser in /tests/UI

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.14.0 to 8.16.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.16.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jolelievre committed:
Update UI testing library

tswfi committed:
If price is not shown hide add to cart too. Fixes #37532

dependabot[bot] committed:
build(deps): bump mysql2 from 3.11.4 to 3.11.5 in /tests/UI

Bumps [mysql2](https://github.com/sidorares/node-mysql2) from 3.11.4 to 3.11.5.
- [Release notes](https://github.com/sidorares/node-mysql2/releases)
- [Changelog](https://github.com/sidorares/node-mysql2/blob/master/Changelog.md)
- [Commits](https://github.com/sidorares/node-mysql2/compare/v3.11.4...v3.11.5)

---
updated-dependencies:
- dependency-name: mysql2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

nicosomb committed:
Update default catalog 9.0.x

Progi1984 committed:
Functional Tests : Removed `jsdoc-to-markdown`

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

Progi1984 committed:
Functional Tests : Audit - BO - Design

dependabot[bot] committed:
build(deps): bump dotenv from 16.4.5 to 16.4.7 in /tests/UI

Bumps [dotenv](https://github.com/motdotla/dotenv) from 16.4.5 to 16.4.7.
- [Changelog](https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md)
- [Commits](https://github.com/motdotla/dotenv/compare/v16.4.5...v16.4.7)

---
updated-dependencies:
- dependency-name: dotenv
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jolelievre committed:
Switch carrier feature flag to stable and enabled by default

Hlavtox committed:
Adjust error handling

dependabot[bot] committed:
build(deps): bump @faker-js/faker from 9.2.0 to 9.3.0 in /tests/UI

Bumps [@faker-js/faker](https://github.com/faker-js/faker) from 9.2.0 to 9.3.0.
- [Release notes](https://github.com/faker-js/faker/releases)
- [Changelog](https://github.com/faker-js/faker/blob/next/CHANGELOG.md)
- [Commits](https://github.com/faker-js/faker/compare/v9.2.0...v9.3.0)

---
updated-dependencies:
- dependency-name: "@faker-js/faker"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jolelievre committed:
Update default catalog 9.0.x

nicosomb committed:
Fixed php cs

Progi1984 committed:
Functional Tests : Bump `@prestashop-core/ui-testing`

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser in /tests/UI

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.19.1 to 8.20.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.20.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @faker-js/faker from 9.3.0 to 9.4.0 in /tests/UI

Bumps [@faker-js/faker](https://github.com/faker-js/faker) from 9.3.0 to 9.4.0.
- [Release notes](https://github.com/faker-js/faker/releases)
- [Changelog](https://github.com/faker-js/faker/blob/next/CHANGELOG.md)
- [Commits](https://github.com/faker-js/faker/compare/v9.3.0...v9.4.0)

---
updated-dependencies:
- dependency-name: "@faker-js/faker"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Progi1984 committed:
`8.2.x` : Use Ubuntu 22.04

Progi1984 committed:
Functional Tests : Bump `@prestashop-core/ui-testing`

mattgoud committed:
fix: ##37873 - downgrade lockfileVersion of themes package-lock.json files

mattgoud committed:
fix: #37886 - [docker-compose] - upgrade node 14 to 16 (with package-lock.json files generated with node 16)

mattgoud committed:
chore: [gitub workflows] upgrade action/{download-upload}-artifact to v4

mattgoud committed:
chore: [release-8.2.1] - update version number to 8.2.1

matthieu-rolland committed:
remove useless try catch that is now useless

matthieu-rolland committed:
update phpdoc

Progi1984 committed:
Functional Tests : Migrate `@pages/BO/advancedParameters/multistore` into `ui-testing-library`

jolelievre committed:
CS fix

jf-viguier committed:
Fix datacollector caster for reference object property

jf-viguier committed:
fix price_final LocalizationException

Progi1984 committed:
Functional Tests: Bump `@prestashop-core/ui-testing`

Codencode committed:
Improved meta tag handling in maintenance mode

Added a check to allow administrators to view the correct meta tags even when the shop is in maintenance mode. Now, if the PS_MAINTENANCE_ALLOW_ADMINS option is enabled and the user is a logged-in administrator, the getMetaTags function will return the appropriate metadata for the requested pages.

boherm committed:
fix product filters for csv export

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser in /tests/UI

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.21.0 to 8.22.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.22.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mattgoud committed:
// Changelog 8.2.1

Codencode committed:
Update Product.php

Changed the selection of the "active" field to take into account the "product_shop" table.

Codencode committed:
Removed duplicate "physical_uri"

In case of installation in a subfolder, "physical_uri" is added twice.

boherm committed:
Change version to 9.1.0

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser in /tests/UI

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.23.0 to 8.24.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.24.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Codencode committed:
Added the "locale" parameter to the method PrestaShopTranslatorTrait::shouldFallbackToLegacyModuleTranslation()

The method PrestaShopTranslatorTrait::shouldFallbackToLegacyModuleTranslation() must take into account the "locale" parameter in order to load the "Catalogue" for the appropriate "locale".

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

jolelievre committed:
Improve how we check if product indexation is needed by checking all updated fields used for the indexation

Progi1984 committed:
Functional Tests : Stabilisation "functional:FO:hummingbird:04-07"

nicosomb committed:
Update package* files

Codencode committed:
Added the 'layoutTitle' variable in the ProductController::renderEditProductForm() method.

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin in /tests/UI

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 8.26.1 to 8.27.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.27.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

fox-john committed:
fix: carrier product delete

Progi1984 committed:
Functional Tests : Bump `@prestashop-core/ui-testing`

unlocomqx committed:
[feat] Add hook actionCartDuplicate to allow modules to duplicate any data that might be related to the old cart

gwojcik7 committed:
add actionProductGetAttributesGroupsAfter hook

gwojcik7 committed:
add actionProductGetAttributesGroupsBefore to manipulate query class

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser in /tests/UI

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.28.0 to 8.29.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.29.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-version: 8.29.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Progi1984 committed:
Functional Tests : Allow override URL_FO for PS Entreprise UI Tests

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

jolelievre committed:
Update ui testing library

nicosomb committed:
fixed trans() params

nicosomb committed:
Fixed package-lock.json

Pliciweb committed:
fix #37354

Codencode committed:
Added check on 'Maximum size for attached files' using Tools::getMaxUploadSize()

This change validates that the 'Maximum size for attached files' does not exceed the server's 'upload_max_filesize' and 'post_max_size' settings.

jf-viguier committed:
Don't delete cart associated to orders, don't send email to deleted customers

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

PrestaEdit committed:
Use v8 documentation in HelperCard

dependabot[bot] committed:
build(deps): bump @faker-js/faker from 9.7.0 to 9.8.0 in /tests/UI

Bumps [@faker-js/faker](https://github.com/faker-js/faker) from 9.7.0 to 9.8.0.
- [Release notes](https://github.com/faker-js/faker/releases)
- [Changelog](https://github.com/faker-js/faker/blob/next/CHANGELOG.md)
- [Commits](https://github.com/faker-js/faker/compare/v9.7.0...v9.8.0)

---
updated-dependencies:
- dependency-name: "@faker-js/faker"
  dependency-version: 9.8.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Progi1984 committed:
Functional Tests : Migrate `@utils/globals` into `ui-testing-library`

ShaiMagal committed:
Update Warehouse.php

ShaiMagal committed:
Update CustomerType.php

dependabot[bot] committed:
build(deps): bump mocha from 11.3.0 to 11.5.0 in /tests/UI

Bumps [mocha](https://github.com/mochajs/mocha) from 11.3.0 to 11.5.0.
- [Release notes](https://github.com/mochajs/mocha/releases)
- [Changelog](https://github.com/mochajs/mocha/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mochajs/mocha/compare/v11.3.0...v11.5.0)

---
updated-dependencies:
- dependency-name: mocha
  dependency-version: 11.5.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Hlavtox committed:
Add hook to apply cart rules

CedricKiwik committed:
PR to hadd hook on Configuration updateValue

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

ludoviccardinale committed:
add 2 hooks in getImageLink

studiokiwik committed:
Fix to pass all parameters

Co-authored-by: Daniel Hlaváček <daniel.hlavacek@hotmail.cz>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-import in /tests/UI

Bumps [eslint-plugin-import](https://github.com/import-js/eslint-plugin-import) from 2.31.0 to 2.32.0.
- [Release notes](https://github.com/import-js/eslint-plugin-import/releases)
- [Changelog](https://github.com/import-js/eslint-plugin-import/blob/main/CHANGELOG.md)
- [Commits](https://github.com/import-js/eslint-plugin-import/compare/v2.31.0...v2.32.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-import
  dependency-version: 2.32.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
BO : Product Form : Use the correct configuration for the Max Filesize

Hlavtox committed:
Speedup getting combination images if product doesn't have any

Venatum committed:
docs(docker): Update default admin credentials in docker-compose.yml and improve README documentation

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin in /tests/UI

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 8.34.0 to 8.35.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.35.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-version: 8.35.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Hlavtox committed:
Wipe @since tags that serve no value anymore

dependabot[bot] committed:
build(deps): bump dotenv from 16.5.0 to 17.0.0 in /tests/UI

Bumps [dotenv](https://github.com/motdotla/dotenv) from 16.5.0 to 17.0.0.
- [Changelog](https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md)
- [Commits](https://github.com/motdotla/dotenv/compare/v16.5.0...v17.0.0)

---
updated-dependencies:
- dependency-name: dotenv
  dependency-version: 17.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Progi1984 committed:
Functional Tests : BO - Cart rules - Actions - Apply a discount percent(%)

Progi1984 committed:
Functional Tests : Bump @prestashop-core/ui-testing

tblivet committed:
fix: preview logo display and error

jolelievre committed:
Remove getCasters override not necessary anymore with SF 6.4

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin in /tests/UI

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 8.35.1 to 8.36.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.36.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-version: 8.36.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

PrestaEdit committed:
Refactor hook registration access and checks

Replaced usage of adminModuleProvider with adminModuleDataProvider for access checks in registerHook and unregisterHook methods. Renamed checkIsInstalled to assertIsInstalled and removed unnecessary cache clearing calls after hook registration and unregistration.

jolelievre commented on src/Adapter/Module/Module.php:
```diff
@@ -46,8 +46,9 @@ class Module implements ModuleInterface
     public const ACTION_RESET = 'reset';
     public const ACTION_UPGRADE = 'upgrade';
     public const ACTION_CONFIGURE = 'configure';
+    public const ACTION_REGISTER_HOOK = 'registerHook';
```
These const are used nowhere for now, but it would be relevant to add them in the `ModuleController::moduleAction` method that checks for authorize permissions. It's probably similar to other actions and would require `$this->isGranted(Permission::UPDATE, self::CONTROLLER_NAME)`

jolelievre replied:
Also, the `AdminModuleDataProvider::isAllowedAccess` method would deserve a little refacto and use these const instead of hrard-coded values, it will help finding where these values are used more easily.

jolelievre commented on src/Core/Module/ModuleInterface.php:
```diff
@@ -76,4 +76,14 @@ public function onReset(): bool;
+    /**
+     * @return bool
+     */
+    public function onRegisterHook(string $hookName): bool;
```
Be careful adding these methods on the interface can create breaking changes, all the classes that implement this interface without these new methods will trigger errors, for a major version we could accept it but for a minor it's more an issue.

To go around this problem you could create another interface, for example `HookModuleInterface` that extends this one and adds these two new methods, the problem is that we start multiplying interfaces when one should be enough... So maybe in a future version we'll have to re-merge them.

Another alternative would be to wait for the next major version to add these two methods on this interface, in the meantime we'll need to check if the module object implements these methods.

PrestaEdit replied:
The PR is a long-time ago created, and it was for a major version.

The only entry having this interface used have already the two methods inside it.

jolelievre commented on src/Core/Module/ModuleManager.php:
```diff
@@ -292,6 +292,66 @@ public function reset(string $name, bool $keepData = false): bool
+     *
+     * @return bool True for success
+     */
+    public function unregisterHook($name, $hooksName)
```
```suggestion
    public function unregisterHook(string $moduleName, array $hookNames)
```

jolelievre commented on src/Core/Module/ModuleManager.php:
```diff
@@ -292,6 +292,66 @@ public function reset(string $name, bool $keepData = false): bool
+     *
+     * @return bool True for success
+     */
+    public function registerHook($name, $hooksName)
```
```suggestion
    public function registerHook(string $moduleName, array $hookNames)
```
I know the other methods use different naming, but it still deserves improvement No nee need to change it everywhere but we can at elast improve it on these new methods

jolelievre commented on src/PrestaShopBundle/Command/ModuleCommand.php:
```diff
@@ -50,6 +51,8 @@ class ModuleCommand extends Command
         'reset',
         'upgrade',
         'configure',
+        'registerHook',
```
We should probably use the consts here as well

PrestaEdit committed:
Update src/Core/Module/ModuleManager.php

Co-authored-by: Jonathan Lelievre <jo.lelievre@gmail.com>

PrestaEdit committed:
Refactor hook registration method parameter names

Renamed parameters in registerHook and unregisterHook methods for clarity and consistency. Updated variable names from $name and $hooksName to $moduleName and $hookNames, and adjusted type hints to improve code readability and maintainability.

Co-Authored-By: Jonathan Lelievre <13801017+jolelievre@users.noreply.github.com>

PrestaEdit committed:
Refactor module actions to use Module class constants

Replaced hardcoded action strings with Module class constants in AdminModuleDataProvider and ModuleCommand. This improves maintainability and reduces the risk of typos or inconsistencies in action names.
